### ISSUE TITLE
Performance: \RestfulManager::invalidateEntityCache calls getAccount(FALSE)

### ISSUE THREAD

**Issue by amitaibu** [RESTful-Drupal/restful#502]
``` php
        // Get the uid for the invalidation.
        try {
          $uid = $handler->getAccount(FALSE)->uid;
        }
```

This means that on every `\RestfulManager::invalidateEntityCache` we re-authenticate the same user. @mateu-aguilo-bosch do you remember the reason for disabling the static cache here?

**Comment by e0ipso** [comment]
Yes, it's because if we don't accept any authenticaction provider, but authentication is optional the expected result is the anon user. If we accept cookie, the expected result is the logged in user. This means different results depending on the plugin definition.
